python/cudf/dataframe/dataframe.py

```diff
@@ -2826,16 +2832,27 @@ def quantile(self,
         DataFrame
 
         """
+        if axis not in (0, None):
+            raise NotImplementedError("axis is not implemented yet")
+
+        if not numeric_only:
+            raise NotImplementedError("numeric_only is not implemented yet")
         if columns is None:
             columns = self.columns
 
         result = DataFrame()
         result['Quantile'] = q
         for k, col in self._cols.items():
             if k in columns:
-                result[k] = col.quantile(q, interpolation=interpolation,
-                                         exact=exact,
-                                         quant_index=False)
+                res = col.quantile(q, interpolation=interpolation,
+                                   exact=exact,
+                                   quant_index=False)
+                if not isinstance(res, (int, float)) and len(res) == 0:
```

Comment:
Need to generalize this as well, also shouldn't `res` already be the correct empty column?

Reply:
Yes, `res` will only have correct empty column with respect to it's column size. But, since `res` is going to be assigned to a column of dataframe, it needs to be of equal dimension to other columns of the data-frame(in this case `'Quantile'`). That's why we will have to create a column of size `q`  and assign.